packages/smooth_app/lib/pages/product/attribute_icons.dart

```diff
@@ -0,0 +1,615 @@
+          backgroundColor: backgroundColor,
+          foregroundColor: foregroundColor,
+          size: size,
+          semanticsLabel: semanticsLabel,
+        ),
+      'vegetarian' => _AttributeVegetarianIcon(
+          backgroundColor: backgroundColor,
+          foregroundColor: foregroundColor,
+          size: size,
+          semanticsLabel: semanticsLabel,
+        ),
+
+      /*'nutriscore' => EMPTY_WIDGET,
+      'ecoscore' => EMPTY_WIDGET,*/
+      _ => throw UnsupportedError('Unsupported attribute: ${attribute.id}'),
+    };
+  }
+
+  const AttributeIcon._({
+    required this.icon,
+    required this.backgroundColor,
+    this.size,
+    this.iconSize,
+    this.angle,
+    this.offset,
+    Color? foregroundColor,
+    this.padding,
+    this.semanticsLabel,
+    this.clip = false,
+  }) : foregroundColor = foregroundColor ?? Colors.white;
+
+  final Widget icon;
+  final Color backgroundColor;
+  final Color foregroundColor;
+  final double? size;
+  final double? angle;
+  final Offset? offset;
+  final double? iconSize;
+  final EdgeInsetsGeometry? padding;
+  final String? semanticsLabel;
+  final bool? clip;
+
+  @override
+  Widget build(BuildContext context) {
+    Widget child = Semantics(
+      label: semanticsLabel,
+      excludeSemantics: true,
+      child: DecoratedBox(
+        decoration: ShapeDecoration(
+          shape: const CircleBorder(),
+          color: backgroundColor,
+        ),
+        child: Padding(
+          padding: padding ?? EdgeInsets.zero,
+          child: Transform.translate(
+            offset: offset ?? Offset.zero,
+            child: AppIconTheme(
+              color: foregroundColor,
+              size: iconSize,
+              child: icon,
+            ),
+          ),
+        ),
+      ),
+    );
+
+    if (angle != null) {
+      child = Transform.rotate(
+        angle: angle!,
+        child: child,
+      );
+    }
+
+    if (clip == true) {
+      child = ClipOval(
+        child: child,
+      );
+    }
+
+    if (size != null) {
+      return SizedBox.square(
+        dimension: size,
+        child: child,
+      );
+    }
+    return child;
+  }
+}
+
+class _AttributeAdditivesIcon extends AttributeIcon {
+  _AttributeAdditivesIcon({
+    required super.backgroundColor,
+    required super.size,
+    super.foregroundColor,
+    super.semanticsLabel,
+  }) : super._(
+          icon: const FoodIcons.additives(),
+          iconSize: size! * 0.6,
+          padding: EdgeInsetsDirectional.only(bottom: size * 0.05),
+        );
+}
+
+class _AttributeCeleryIcon extends AttributeIcon {
+  _AttributeCeleryIcon({
+    required super.backgroundColor,
+    required super.size,
+    super.foregroundColor,
+    super.semanticsLabel,
+  }) : super._(
+          icon: const FoodIcons.celery(),
+          iconSize: size! * 0.95,
+          offset: Offset(size * -0.1, 0.0),
+          clip: true,
+          padding: EdgeInsetsDirectional.only(top: size * 0.15),
+        );
+}
```

Comment:
All that is quite verbose and redundant.
Would be easier to read with optional parameters stored at the FontIconFonts level, like:
* sizeFactor: .95
* clip: true
* paddingFactor: Rect.fromLTRB(0, .15, 0, 0)